rest/src/main/java/org/n52/web/ctrl/ParameterController.java

```diff
@@ -132,22 +135,48 @@ public void getRawData(HttpServletResponse response,
     }
 
     @Override
-    public ModelAndView getCollection(String locale, MultiValueMap<String, String> query) {
+    public ModelAndView getCollection(HttpServletRequest request, HttpServletResponse response, String locale, MultiValueMap<String, String> query) {
         RequestUtils.overrideQueryLocaleWhenSet(locale, query);
         IoParameters queryMap = QueryParameters.createFromQuery(query);
         LOGGER.debug("getCollection() with query '{}'", queryMap);
-
+        OutputCollection<T> result;
+        
         if (queryMap.isExpanded()) {
             Stopwatch stopwatch = Stopwatch.startStopwatch();
-            OutputCollection<T> result = addExtensionInfos(parameterService.getExpandedParameters(queryMap));
+            result = addExtensionInfos(parameterService.getExpandedParameters(queryMap));
             LOGGER.debug("Processing request took {} seconds.", stopwatch.stopInSeconds());
-            return createModelAndView(result);
         } else {
-            OutputCollection<T> results = parameterService.getCondensedParameters(queryMap);
-            return createModelAndView(results);
+            result = parameterService.getCondensedParameters(queryMap);
+        }
+        
+        // Add Paging if requested
+        if (queryMap.containsParameter("limit") || queryMap.containsParameter("offset")){
+            Paginated<T> paginated = new Paginated(new OffsetBasedPagination(queryMap.getOffset(), queryMap.getLimit()), result.size());
+            this.addPagingHeaders(request.getRequestURI(), response, paginated);
```

Comment:
here you should be careful, as `request.getRequestURI()` does not give you the /real/ URI the client might have called (think of a proxied environment). The external URL (configured in `application.properties`) can be obtained via `IoParameters.map.getHrefBase()`.

However, you've to know the actual collection to create the right link, .e.g `http://localhost:8080/webapp-context/api/datasets` rather. For this, you could add an abstract method `getCollectionPath()` all parameter controllers have to implement and construct the full URL via [UrlHelper#constructHref(String, String)](https://github.com/52North/series-rest-api/blob/develop/rest/src/main/java/org/n52/web/ctrl/UrlHelper.java#L69)